velero-plugins/route/restore.go

```diff
@@ -31,26 +31,32 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	route := routev1API.Route{}
 	itemMarshal, _ := json.Marshal(input.Item)
 	json.Unmarshal(itemMarshal, &route)
-
-	client, err := clients.CoreClient()
-	if err != nil {
-		return nil, err
-	}
-	config, err := client.ConfigMaps("openshift-apiserver").Get("config", metav1.GetOptions{})
-	if err != nil {
-		return nil, err
-	}
-	serverConfig := common.APIServerConfig{}
-	err = json.Unmarshal([]byte(config.Data["config.yaml"]), &serverConfig)
+	version, err := common.GetServerVersion()
 	if err != nil {
 		return nil, err
 	}
+	if version.Major == "4" {
+		client, err := clients.NewCoreClient()
+		if err != nil {
+			return nil, err
+		}
+		config, err := client.ConfigMaps("openshift-apiserver").Get("config", metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		serverConfig := common.APIServerConfig{}
+		err = json.Unmarshal([]byte(config.Data["config.yaml"]), &serverConfig)
+		if err != nil {
+			return nil, err
+		}
 
-	subdomain := serverConfig.RoutingConfig.Subdomain
+		subdomain := serverConfig.RoutingConfig.Subdomain
 
-	output := replaceSubdomain(input.Item, &route, subdomain)
+		output := replaceSubdomain(input.Item, &route, subdomain)
+		return output, nil
+	}
 
-	return output, nil
+	return nil, nil
```

Comment:
Also, it looks like in the 3.11 case, we're not replacing anything. Don't we still need to fix the subdomain for 3.x, since it still might be a different cluster than the source.

Reply:
Yes we do. Unfortunately there is no way to read the master-config file from our plugin to determine the proper subdomain. So for 3.x the safest approach in my view is to leave the route the same, which I have reflected in my recent commit.